packages/qb/pqb/src/queryMethods/transaction.test.ts

```diff
@@ -237,4 +237,128 @@ describe('transaction', () => {
       });
     });
   });
+
+  describe('afterCommit', () => {
+    afterEach(() => {
+      const t = testDb.internal.transactionStorage.getStore();
+      if (t) {
+        delete t.afterCommit;
+        delete t.catchAfterCommitError;
+      }
+    });
+
+    // Async hook is a jest mock which only records the call only after it has been awaited.
+    const createAsyncHook = () => {
+      const mock = jest.fn();
+      return new Proxy(mock, {
+        apply: async (...args) => {
+          await new Promise((resolve) => setTimeout(resolve, 10));
```

Comment:
1. This can be converted to `await new Promise(setImmediate)` which also works but will catch less bugs. Probably enough for this case though.

3. This can actually be skipped completely (simply use `jest.fn()`), but then there is no check that the hook is awaited (only that it's called).